Sources/StreamChatUI/Utils/LoaderView.swift

```diff
@@ -0,0 +1,307 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+extension UIView {
+    func showLoader(heightToCornerRadiusRatio: CGFloat) {
+        isUserInteractionEnabled = false
+        if self is UITableView {
+            ListLoader.addLoaderTo(self as! UITableView, heightToCornerRadiusRatio: heightToCornerRadiusRatio)
+        } else if self is UICollectionView {
+            ListLoader.addLoaderTo(self as! UICollectionView, heightToCornerRadiusRatio: heightToCornerRadiusRatio)
+        } else {
+            ListLoader.addLoaderToViews([self], heightToCornerRadiusRatio: heightToCornerRadiusRatio)
+        }
+    }
+    
+    func hideLoader() {
+        isUserInteractionEnabled = true
+        if self is UITableView {
+            ListLoader.removeLoaderFrom(self as! UITableView)
+        } else if self is UICollectionView {
+            ListLoader.removeLoaderFrom(self as! UICollectionView)
+        } else {
+            ListLoader.removeLoaderFromViews([self])
+        }
+    }
+}
+
+protocol ListLoadable {
+    func ld_visibleContentViews() -> [UIView]
+}
+
+extension UITableView: ListLoadable {
+    func ld_visibleContentViews() -> [UIView] {
+        (visibleCells as NSArray).value(forKey: "contentView") as! [UIView]
+    }
+}
+
+extension UICollectionView: ListLoadable {
+    func ld_visibleContentViews() -> [UIView] {
+        (visibleCells as NSArray).value(forKey: "contentView") as! [UIView]
```

Comment:
sometimes UIKit work behind the hood changes, so I would recommend to use safe version:
`(visibleCells as NSArray).value(forKey: "contentView") as? [UIView] : []`